src/fileimpl.cpp

```diff
@@ -69,16 +69,18 @@ struct NoDelete
 };
 
 std::shared_ptr<Reader>
-makeFileReader(std::shared_ptr<const FileCompound> zimFile)
+makeFileReader(std::shared_ptr<const FileCompound> zimFile, offset_t offset, zsize_t size)
 {
   if (zimFile->fail())
     return nullptr;
-  else if ( zimFile->is_multiPart() )
+  else if ( zimFile->is_multiPart() ) {
+    ASSERT(offset.v, ==, 0u);
+    ASSERT(size, ==, zimFile->fsize());
     return std::make_shared<MultiPartFileReader>(zimFile);
-  else {
+  } else {
```

Comment:
I missed this in previous commit. But I tend to have braces all the time with `if`/`else`/`while`/... 

It avoid potential future bugs and simplify changes (this one would be simple add of two lines instead of a change of four lines)